src/app_model/registries/_keybindings_reg.py

```diff
@@ -38,9 +64,11 @@ class KeyBindingsRegistry:
     """
 
     registered = Signal()
+    unregistered = Signal()
 
     def __init__(self) -> None:
         self._keybindings: list[_RegisteredKeyBinding] = []
+        self._keymap: dict[int, list[_RegisteredKeyBinding]] = {}
```

Comment:
```suggestion
        self._keymap = defaultdict[int, list[_RegisteredKeyBinding]](list)
```

and import defaultdict from collections

Reply:
Also, I don't think we need to maintain *both* `_keybindings` and `_keymap` do we?  I believe you could just delete `self._keybindings` here, get rid of the `_keybindings.append` and `_keybindings.remove` logic in `register...`, and replace it with a property that just iterates through all the lists in `_keymap`:

```python
    @property
    def _keybindings(self) -> Iterable[_RegisteredKeyBinding]:
        return (entry for entries in self._keymap.values() for entry in entries)
```

then you'll also need to remove `len(self._keybindings)` from `__repr__` ... and could use `len(self)` instead with:

```python
    def __len__(self) -> int:
        return sum(len(entries) for entries in self._keymap.values())
```